tools/tests/test_validate_sch.py

```diff
@@ -0,0 +1,166 @@
+import os
+import sys
+
+from lxml import etree
+import pytest
+
+from tools.validate_sch import validate_schematron
+
+@pytest.fixture
+def simple_valid_doc_content():
+    return '''<root>
+        <child attr="hello"/>
+    </root>'''
+
+@pytest.fixture
+def simple_bad_doc_content():
+    return '''<root>
+        <child attr="world"/>
+    </root>'''
+
+@pytest.fixture
+def simple_sch_content():
+    return '''
+    <sch:schema xmlns:sch="http://purl.oclc.org/dsdl/schematron">
+        <sch:pattern>
+            <sch:rule context="/root/child">
+                <sch:assert test="@attr = 'hello'" role="ERROR">Attr should be hello</sch:assert>
+            </sch:rule>
+        </sch:pattern>
+    </sch:schema>'''
+
+class TestValidateSchematron:
+    def test_when_doc_is_valid_returns_no_errors(self, simple_valid_doc_content, simple_sch_content):
+        # -- Act
+        failures = validate_schematron(simple_sch_content, simple_valid_doc_content)
+
+        # -- Assert
+        assert len(failures) == 0
+
+
+    def test_accepts_file_path_or_str_content(self, tmpdir, simple_valid_doc_content, simple_sch_content):
+        # -- Setup
+        doc = os.path.join(tmpdir, "test.xml")
+        with open(doc, 'w') as f:
+            f.write(simple_valid_doc_content)
+        sch = os.path.join(tmpdir, "test.sch")
+        with open(sch, 'w') as f:
+            f.write(simple_sch_content)
+        
+        # -- Act
+        failures = validate_schematron(sch, doc)
```

Comment:
doesn't this only test in the event that it is a file path?  don't we need another `validate_schematron`?

Reply:
updated test name 👍